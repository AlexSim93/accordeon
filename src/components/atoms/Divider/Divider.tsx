import styled from 'styled-components';
import { layout, color } from 'styled-system';

export const Divider = styled.div<any>`
    ${layout};
    ${color}
`;
