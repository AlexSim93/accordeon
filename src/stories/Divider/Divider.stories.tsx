import React from 'react';
import { storiesOf } from '@storybook/react';
import { Box, Flex } from 'reflexbox';
import { ThemeProvider } from 'styled-components';
import { Divider } from '../../components/atoms/Divider';
import notes from './readme.md';
import { Text } from '../../components/atoms/Text';
import theme from '../../theme';

storiesOf('Divider', module).add('Divider', () => (
  <ThemeProvider theme={theme}>
    <Flex flexDirection="column">
      <Text color="black" fontWeight="600" fontSize="xl">Divider</Text>
      <Text color="black" fontSize="l">Divider with height=10 and background=black</Text>
      <Box>
        <Divider height={10} bg="black" />
      </Box>
    </Flex>
  </ThemeProvider>
), { notes });
