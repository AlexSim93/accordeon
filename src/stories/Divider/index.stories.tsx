import React from 'react';
import { storiesOf } from '@storybook/react';
import { Box } from 'reflexbox';
import { Divider } from '../../components/atoms/Divider';
import notes from './readme.md';

storiesOf('Divider', module).add('Divider', () => (
  <Box>
    <Box>Divider</Box>
    <Box>height=10; bg=#080D26</Box>
    <Box>
      <Divider height={10} bg="#080D26" />
    </Box>
  </Box>
)).addParameters({ notes });
