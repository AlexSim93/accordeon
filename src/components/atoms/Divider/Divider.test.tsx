import React from 'react';
import renderer from 'react-test-renderer';
import { Divider } from './Divider';

describe('Divider renders correctly', () => {
    test('Divider with height=10 and bg="black" renders correctly', () => {
        const tree = renderer
            .create(<Divider height={10} bg="black" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})