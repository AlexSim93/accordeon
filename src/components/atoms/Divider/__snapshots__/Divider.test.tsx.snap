// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly Divider with height=10 and bg="black" renders correctly 1`] = `
<div
  className="sc-AxjAm fEbVWD"
  height={10}
/>
`;
